tests/h/services/search_index/service_test.py

```diff
@@ -152,6 +153,87 @@ def test_it_calls_elasticsearch_again_for_a_reindex(
         )
 
 
+class TestHandleAnnotationEvent:
+    def test_we_dispatch_correctly(
+        self, search_index, pyramid_request, action, synchronous, handler_for
+    ):
+        event = AnnotationEvent(pyramid_request, {"id": "any"}, action)
+
+        result = search_index.handle_annotation_event(event, synchronous=synchronous)
+
+        handler = handler_for(action, synchronous)
+        handler.assert_called_once_with(event.annotation_id)
+        assert result == handler.return_value
+
+    def test_we_do_nothing_for_unexpected_actions(
+        self, search_index, pyramid_request, synchronous
+    ):
+        event = AnnotationEvent(pyramid_request, {"id": "any"}, "strange_action")
+
+        result = search_index.handle_annotation_event(event, synchronous=synchronous)
+
+        assert result is False
```

Comment:
This could maybe also do `handler.assert_not_called()` for good measure

Reply:
There is no handler in this case, so we'd have to iterate over all of them, which sounds like a pain.